### PR TITLE
Blog post for 0.8.15

### PR DESCRIPTION
**Pull request by cameel**

### CODE REVIEW

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+---
+
+[Solidity v0.8.15](https://github.com/ethereum/solidity/releases/tag/v0.8.15)
+fixes two important bugs, improves inlining heuristics and adds ``.selector`` member for errors and events.
+
+## Important Bugs
+
+The first one is an optimizer bug that can lead to memory write operations in inline assembly being discarded if
+the result of such an operation is not read back from within the same assembly block.
+The bug can be triggered only when not using the old, non-IR compiler pipeline and happens only in assembly blocks
```
Comment (ekpyron) [inline _posts/2022-06-15-solidity-0.8.15-release-announcement.md:17]:
```suggestion
The bug can be triggered only when using the old, non-IR compiler pipeline and happens only in assembly blocks
```

Or maybe
```suggestion
The bug can be triggered only when not using the default legacy compiler pipeline (the new compilation pipeline via IR is not affected) and happens only in assembly blocks
```

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+### Improved Inlining Heuristics in Yul Optimizer
+
+Until now the the compiler used to be very conservative in deciding whether to inline a function or not.
+This was necessary due to the fact that inlining may easily increase stack pressure and lead to the dreaded "Stack too deep" error.
+
+Now that the mechanism for moving variables from stack to memory is available for the IR pipeline, we were able to relax the
+conditions necessary for inlining.
+The optimizer will now inline slightly bigger functions as long as nothing prevents the mechanism from being used.
+If you want to benefit from it, make sure that your inline assembly blocks are [memory safe](https://docs.soliditylang.org/en/latest/assembly.html#memory-safety)
+conditions, and use an EVM version newer than Homestead as the compilation target.
```
Comment (ekpyron) [inline _posts/2022-06-15-solidity-0.8.15-release-announcement.md:37]: I'm not sure whether we need to explicitly mention homestead here - I'd imagine that people really do that anyways.

Comment (cameel) [reply _posts/2022-06-15-solidity-0.8.15-release-announcement.md]: Fair point. Changed.

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md (hunk 1 above)
Comment (cameel) [reply _posts/2022-06-15-solidity-0.8.15-release-announcement.md]: Done using the second version.

**Comment by cameel** [comment]
I just pushed some more small tweaks that I was working on while you were reviewing. Mostly in the section about the selectors.

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+layout: post
+published: true
+title: 'Solidity 0.8.15 Release Announcement'
+date: '2022-06-15'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.15](https://github.com/ethereum/solidity/releases/tag/v0.8.15)
+fixes two important bugs, improves inlining heuristics and adds the ``.selector`` member for errors and events.
```
Comment (ekpyron) [inline _posts/2022-06-15-solidity-0.8.15-release-announcement.md:11]: Arguably
```suggestion
fixes two important bugs, improves inlining heuristics and adds a ``.selector`` member for errors and events.
```

**Comment by cameel** [comment]
I added a fragment encouraging people to upgrade from 0.8.13 and 0.8.14.

**Comment by matheusaaguiar** [comment]
Ok, just missing the full changelog part.

**Comment by cameel** [comment]
Changelog added.

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+Note that for events the selector represents the topic 0, which, unlike error and function selectors, is not truncated to 4 bytes.
+
+## New or Improved Documentation
+
+### Dangling References
+
+We have recently received vulnerability reports about cases where normal use of language features may leave the user with
+a dangling reference and lead to data being overwritten in storage.
+Unfortunately this is a natural consequence of the current design of the language and not a bug that can be simply fixed.
```
Comment (ekpyron) [inline _posts/2022-06-15-solidity-0.8.15-release-announcement.md:59]:
```suggestion
Unfortunately, this is a natural consequence of the current design of the language and not a bug that can be simply fixed.
```

**Review by ekpyron — APPROVED** [review]
Looks good apart from minor comment.

File: _posts/2022-06-15-solidity-0.8.15-release-announcement.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+## New or Improved Documentation
+
+### Dangling References
+
+We have recently received vulnerability reports about cases where normal use of language features may leave the user with
+a dangling reference and lead to data being overwritten in storage.
+Unfortunately, this is a natural consequence of the current design of the language and not a bug that can be simply fixed.
+Solidity severely restricts the use of references to value types, but the availability of nested arrays (which are reference
+types) combined with the ability to resize them has always meant that the target of a reference may disappear.
```
[the remark this answers is not in the document]
Comment (cameel) [reply _posts/2022-06-15-solidity-0.8.15-release-announcement.md]: Sounds good. Done.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**